### PR TITLE
Add new category filters

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What is the purpose of this pull request?
Change the category filters from being a checkbox list to a tree-like structure (showing 1 children down max).

#### What problem is this solving?
With the checkboxes, we didn't had any guards against a user deselecting a department category on a subcategory page, and that made the product search break, bringing every product to the gallery.

#### How should this be manually tested?
[storecomponents wks](https://lucas--storecomponents.myvtex.com/clothing?map=c).
[invicta wks](https://lucas--invictastores.myvtex.com/watches?map=c).

#### Screenshots or example usage
image

inspired by [flipkart's own search](https://www.flipkart.com/search?q=shoes&otracker=search&otracker1=search&marketplace=FLIPKART&as-show=off&as=off):

image

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [x] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/hooks/useFacetNavigation.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+              .map(decodeURIComponent)
+              .findIndex(
+                value => value === decodeURIComponent(facet.value).toLowerCase()
+              )
+
+            return {
+              currentQuery: removeElementAtIndex(currentQuery, facetIndex, '/'),
+              currentMap: removeElementAtIndex(currentMap, facetIndex, ','),
+            }
+          } else {
```
Comment (jgfidelis) [inline react/hooks/useFacetNavigation.js]: useless else

Comment (lucasecdb) [reply react/hooks/useFacetNavigation.js]: I think it reads better this way but oh well

File: react/searchResult.css [hunk 2]
```diff
@@ -43,10 +43,6 @@
   justify-self: stretch;
 }
 
-.filter span:hover {
```
Comment (klzns) [inline react/searchResult.css]: Keep class `filter`.

File: react/components/CategoryFilters.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        <span
+          className={classNames(
+            styles.filterTitle,
+            't-heading-6 flex items-center justify-between'
+          )}
+        >
+          <FormattedMessage id={title} />
+        </span>
+      </div>
+      {selectedCategories.length > 1 &&
```
[the remark this answers is not in the document]
Comment (lucasecdb) [reply react/components/CategoryFilters.js]: oh, nice!

File: react/components/CategoryItem.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  const navigateToFacet = useFacetNavigation()
+
+  if (shallow) {
+    return (
+      <span
+        role="button"
+        tabIndex={0}
+        className="dib f5 pointer pv2"
+        onClick={() => navigateToFacet(category)}
+        onKeyDown={e => e.key === 'Enter' && navigateToFacet(category)}
```
[the remark this answers is not in the document]
Comment (lucasecdb) [reply react/components/CategoryItem.js]: do you think it's necessary? this constant will *never* change

File: react/hooks/useMeasure.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { useState, useEffect } from 'react'
+import ResizeObserver from 'resize-observer-polyfill'
+
+export default function useMeasure(ref) {
```
Comment (lucasecdb) [inline react/hooks/useMeasure.js]: basically copy and pasted from here https://codesandbox.io/s/q3ypxr5yp4?from-embed

File: react/components/CategoryFilter.js [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+import Collapsible from './Collapsible'
+import CategoryItem from './CategoryItem'
+
+import styles from '../searchResult.css'
+
+const getSelectedCategories = rootCategory => {
+  let node = rootCategory.children
+  const selectedCategories = [rootCategory]
+
+  while (node) {
```
Comment (matheusps) [inline react/components/CategoryFilter.js]: You could use `some` or `every` here. (nit, actually)

File: react/components/CategoryFilter.js [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+      handleUnselectCategories(0)
+    }
+  }
+
+  return (
+    <div className="mt4">
+      <div
+        role="button"
+        tabIndex={canDisableRoot ? 0 : -1}
+        className={classNames('flex items-center pointer')}
```
Comment (matheusps) [inline react/components/CategoryFilter.js]:
```suggestion
        className="flex items-center pointer"
```

Comment (matheusps) [reply react/components/CategoryFilter.js]: nit

File: react/components/Collapsible.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
+import classNames from 'classnames'
+import React, { useState, useRef } from 'react'
+import { injectIntl } from 'react-intl'
+import { useSpring, animated } from 'react-spring'
```
Comment (matheusps) [inline react/components/Collapsible.js]: Your animation is really simple. Could you try it without adding another lib? 🤔

File: react/components/Collapsible.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+}) => {
+  const [open, setOpen] = useState(false)
+  const shouldCollapse = items.length >= maxItems + threshold
+
+  const containerRef = useRef(null)
+
+  const overflowQuantity = items.length - maxItems
+
+  const { height } = useMeasure(containerRef)
+  const styles = useSpring({ height: open ? height : 0 })
```
Comment (matheusps) [inline react/components/Collapsible.js]: Couldn't you do this with `translateY`to avoid the reflow?

**Pull request merged**